kong/clustering/wrpc_check.lua

```diff
@@ -0,0 +1,39 @@
+local http = require "resty.http"
+
+--- detect '/v1/wrpc' endpoint
+--- if there is no '/v1/wrpc', fallback to websocket + json
+local function check_wrpc_support(conf, cert, cert_key)
+  local params = {
+    scheme = "https",
+    method = "HEAD",
+
+    ssl_verify = true,
+    ssl_client_cert = cert,
+    ssl_client_priv_key = cert_key,
+  }
+
+  if conf.cluster_mtls == "shared" then
+    params.ssl_server_name = "kong_clustering"
+
+  else
+    -- server_name will be set to the host if it is not explicitly defined here
+    if conf.cluster_server_name ~= "" then
+      params.ssl_server_name = conf.cluster_server_name
+    end
+  end
+
+  local c = http.new()
+  local res, err = c:request_uri(
+    "https://" .. conf.cluster_control_plane .. "/v1/wrpc", params)
+  if not res then
+    return nil, err
+  end
+
+  if res.status == 404 then
+    return "v0"
```

Comment:
This is a bit strange, the function is called `check_wrpc_support` and one would not expect the return to be `v0` or `v1`. Maybe a simple `true`/`false` would be enough, or the function should be called something like `check_protocol_support` instead.

Reply:
OK, I rename it to `check_protocol_support`.

Reply:
OK, I rename it to `check_protocol_support`.